### PR TITLE
Added Auth fields

### PR DESCRIPTION
**Pull request by pielonet**
Hi,

I added autentication fields to nextcloud/external.

These authentication fields allow opening external sites protected by a login/password form.

It uses an ajax post request made to an authentication url. Once the authentication is successful and the session cookie has been set, the iframe src attribute is changed to match the external site final url.

Since most of the time the ajax request is performed against an other domain than Nextcloud's one we are confronted to cross sites control restrictions. Thus it is necessary to add Access-Control-Allow-xxxx headers to the web server hosting the external site. The most common necessary headers are those : 
Access-Control-Allow-Origin : "https://external.site.domain"
Access-Control-Allow-Methods : "GET, POST, OPTIONS"
Access-Control-Allow-Credentials: "true"
Access-Control-Allow-Headers: "Authorization, Content-Type"

The fourth field I added is there for supplementary headers sent with the ajax request that might be required by the login form.

I successfully used these auth fields to authenticate against Kibana free edition with x-pack security enabled.

Thank you for taking my contribution into account.

Best regards,

Thierry

### CODE REVIEW

**Comment by pielonet** [comment]
Hi,
Anybody out there ?
Still waiting for a review !
Thanks,
Thierry

**Comment by DanielLester83** [comment]
I literally just made a middle man PHP page to do a much simpler version of this. Maybe this code can be of some use. It converts "GET" URL variables to "POST" so that {displayname} anf hopefully a future {password} can be used to automatically login nextcloud users into other websites. This is obviously much less a complete than what you have done, congratulations.

HTML Code:
[usrlogin.php.txt](https://github.com/nextcloud/external/files/3685874/usrlogin.php.txt)

**Comment by pielonet** [comment]
Hi,
@nickvergessen  @MariusBluem  Can you review ?
Thanks,

**Comment by nickvergessen** [comment]
Yeah sorry, it's still on my todo, but I have a lot to do lately

File: l10n/fr.json [hunk 1]
```diff
@@ -1,6 +1,11 @@
 { "translations": {
     "Name" : "Nom",
     "URL" : "URL",
+    "Auth" : "Authorisation",
```
Comment (andrenasturas) [inline l10n/fr.json:4]:
```suggestion
    "Auth" : "Autorisation",
```
Or maybe "Authentification" ?

File: l10n/en_GB.json [hunk 2]
```diff
@@ -1,7 +1,12 @@
 { "translations": {
     "Name" : "Name",
     "URL" : "URL",
+    "Auth" : "Auth",
```
Comment (andrenasturas) [inline l10n/en_GB.json:4]:
```suggestion
    "Auth" : "Authorization",
```
For coherence with i18n/en_GB.js

**Comment by Ornias1993** [comment]
Love the idea of this, but we might want to add a "logout redirect" to the main Nextcloud codebase first (which has been long due anyway), to also enable people to seemlessly log out of an external service on nextcloud logout too.

**Comment by nickvergessen** [comment]
I don't think this is something we want. You can check if #310 solves your issue.

**Pull request closed without merging**